### PR TITLE
supporting pandas on osx, added test suite

### PR DESCRIPTION
**Pull request by adnelson**
- Adds a few additional dependencies to pandas (scipy, numexpr, bottleneck etc) which are listed as optional but are useful to have, providing either performance gains or additional functionality. 
- Can now support OSX as a platform
- Running the test suite, minus a few problematic modules.

### CODE REVIEW

**Comment by adnelson** [comment]
@domenkozar @7c6f434c

File: pkgs/top-level/python-packages.nix [hunk 1]
```diff
@@ -8100,19 +8100,67 @@ let
...
+      platforms = platforms.unix;
+    };
+  };
+
+  xlrd = buildPythonPackage rec {
+    name = "xlrd-${version}";
+    version = "0.9.3";
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/x/xlrd/xlrd-${version}.tar.gz";
+      md5 = "6f3325132f246594988171bc72e1a385";
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: Please don't use md5, but sha256

File: pkgs/top-level/python-packages.nix [hunk 2]
```diff
@@ -8100,19 +8100,67 @@ let
...
+    };
+  };
+
+  bottleneck = buildPythonPackage rec {
+    name = "Bottleneck-${version}";
+    version = "1.0.0";
+    src = pkgs.fetchurl {
+      url = "https://pypi.python.org/packages/source/B/Bottleneck/Bottleneck-${version}.tar.gz";
+      md5 = "380fa6f275bd24f27e7cf0e0d752f5d2";
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: Same here

**Comment by adnelson** [comment]
@domenkozar updated

**Comment by adnelson** [comment]
@domenkozar any thoughts?

File: pkgs/top-level/python-packages.nix [hunk 3]
```diff
@@ -8100,21 +8100,69 @@ let
...
+      echo $first >> pandas/tests/test_format.py
+      echo "$rest" >> pandas/tests/test_format.py
+    '';
+
+    checkPhase = ''
+      runHook preCheck
+
+      # The flag `-A 'not network'` will disable tests that use internet.
+      # The `-e` flag disables a few problematic tests.
+      python setup.py nosetests -A 'not network' --stop -e 'test_clipboard|test_series' --verbosity=3
```
Comment (domenkozar) [inline pkgs/top-level/python-packages.nix]: `python` should be `${python.executable}`

Comment (adnelson) [inline pkgs/top-level/python-packages.nix]: you got it

**Comment by adnelson** [comment]
@domenkozar sorry to keep pressing on this but it'd be nice to get this in

**Pull request merged**